command/agent/command.go

```diff
@@ -73,9 +73,10 @@ func (c *Command) readConfig() *Config {
 		Server: &ServerConfig{
 			ServerJoin: &ServerJoin{},
 		},
-		Vault: &config.VaultConfig{},
-		ACL:   &ACLConfig{},
-		Audit: &config.AuditConfig{},
+		Vault:  &config.VaultConfig{},
+		Vaults: map[string]*config.VaultConfig{},
```

Comment:
Yes, but that also makes me realize that if someone passes the flags they're being set on that `Vault` (which gets overwritten) and not the `Vaults`. So that's a bug to fix.

Reply:
I haven't checked, but the [`vault` configuration reload](https://developer.hashicorp.com/nomad/docs/configuration/vault#vault-configuration-reloads) may need to be adjusted as well:

Reply:
Good looking out. The way the reload works currently is that we call `SetConfig` on the Vault client with the new default Vault config. So until we have those multiple Vault clients instantiated (which will come in a later PR), there's nothing to do for the reload. I'll revisit this then.

Reply:
Fixed in 3dcd0a7913d341bb6bbde1018202bf274c9f3173